### ISSUE TITLE
Strange behaviour in RoomPluginMain.OnData

### ISSUE THREAD

**Issue by Dakkers** [DarkFlamex1/RoomPlugin#3]
Hi! I stumbled across this plugin on one of the DarkRift threads on Unity's forums.

In RoomPluginMain.cs, the `OnData` method has some probably-unwanted behaviour. The logic is something like:

```
if (data.tag == ROOM_CREATE) {
    ....
}
if (data.tag == ROOM_JOIN) {
    ...
}
if (data.tag == ROOM_LEAVE) {
    ...
}
else {
     ...
}
```

so in the scenario where I send data with the tag `ROOM_CREATE`, it will execute that branch, which is fine. But, it will also check the next two `if` conditions, because they aren't `else if`. Both of these will fail, but then the `else` branch will be executed because it is attached to the `ROOM_LEAVE` check, which is `false` in this case.

Is this the intended behaviour?

**Comment by Dakkers** [comment]
Also - if you are open to outside contributions, I am willing to help out. I need a room system for my game as well. Maybe I could add additional functionality or tests or something like that.

**Comment by DarkFlamex1** [comment]
That is not the intended behaviour! Thanks for letting me know! And yes I am open to outside contributions, just use pull requests to commit your code, Nice to see other people actually using this. 👍
If you made fixes just make a pull request cause I am sort of busy the next 2 days.

**Comment by Dakkers** [comment]
closed with #4
